packages/docusaurus/src/server/moduleShorthand.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+export function getNamePatterns(
+  moduleName: string,
+  moduleType: 'preset' | 'theme' | 'plugin',
+): string[] {
+  if (moduleName.startsWith('@')) {
+    // Pure scope: `@scope` => `@scope/docusaurus-plugin`
+    if (!moduleName.includes('/')) {
+      return [`${moduleName}/docusaurus-${moduleType}`];
+    }
+    const [scope, packageName] = moduleName.split(/\/(.*)/);
+    return [
+      `${scope}/${packageName}`,
+      `${scope}/docusaurus-${moduleType}-${packageName}`,
+    ];
+  }
+  return [
+    moduleName,
+    `@docusaurus/${moduleType}-${moduleName}`,
+    `docusaurus-${moduleType}-${moduleName}`,
+  ];
+}
+
+export function resolveModuleName(
+  moduleName: string,
+  moduleRequire: NodeRequire,
+  moduleType: 'preset' | 'theme' | 'plugin',
+): string {
+  const modulePatterns = getNamePatterns(moduleName, moduleType);
+  // eslint-disable-next-line no-restricted-syntax
+  for (const module of modulePatterns) {
+    try {
+      moduleRequire.resolve(module);
```

Comment:
ok for that.

Throwing exceptions has a runtime cost, but it's probably not very important here.
I don't know if there's a fail-safe way to resolve a module in nodejs but if there is one, that would be preferable.